packages/aws-library/src/aws_library/s3/_client.py

```diff
@@ -317,6 +318,60 @@ async def _list_all_objects(
             for obj in s3_objects:
                 yield obj
 
+    @s3_exception_handler_async_gen(_logger)
+    async def list_entries_paginated(
```

Comment:
where is the test for this new function?

Reply:
coming